### PR TITLE
Use modern Signer API in Metablock.verify_signature

### PR DESCRIPTION
**Pull request by lukpueh**
Replace usage of legacy `securesystemslib.keys.verify_signature` with modern verification code from Signer API.

The patch is a pure refactor and does not change behavior. It is part of a series of patches to prepare for the removal of legacy securesystemslib modules.

### CODE REVIEW

File: in_toto/models/metadata.py [hunk 1]
```diff
@@ -395,18 +399,28 @@ def verify_signature(self, verification_key):
...
-        elif securesystemslib.formats.SIGNATURE_SCHEMA.matches(signature):
-            valid = securesystemslib.keys.verify_signature(
-                verification_key, signature, self.signed.signable_bytes
-            )
-
         else:
-            valid = False
+            # Parse key and (below) signature dicts as `Key` and `Signature`
+            # instances to use modern securesystemslib verification code.
+            # Deepcopy to preserve original dicts, which are otherwise destroyed
```
Comment (trishankatdatadog) [inline in_toto/models/metadata.py:411]: Hmm, interesting. Do the `from_dict` methods _need_ to be destructive?

**Review by trishankatdatadog — APPROVED** [review]
LGTM FWIW

File: in_toto/models/metadata.py (hunk 1 above)
Comment (lukpueh) [reply in_toto/models/metadata.py:411]: This aligns with usage in python-tuf's Metadata API, where dicts can be discarded once deserialised.

There it turned out useful to just pop all expected fields and cram whatever's left into  a `unrecognized_fields: Optional[Dict[str, Any]]` attribute, which exists on all classes that have a `from_dict` deserialisation helper.

Unfortunately, this doesn't work in in-toto, where we only have classes for parts of the metadata but keep most contained data (signatures, keys, etc.) as dicts.

That's what I meant with supporting Key and Signature natively being nice. I don't think it's a priority, but I am happy to submit an issue.

**Comment by lukpueh** [comment]
Thanks for the review, @trishankatdatadog! I appreciate you taking the time. :)

File: in_toto/models/metadata.py (hunk 1 above)
Comment (trishankatdatadog) [reply in_toto/models/metadata.py:411]: Oh, I think I see, hmm. Still don't understand why we can't hide away destruction (e.g., by copying) on behalf of users, but you're right, this is not a high-priority issue, although we should make an issue just to be safe. Thanks!

**Pull request merged**